core/modules/query/TableRef.h

```diff
@@ -110,6 +110,16 @@ class TableRef {
     void apply(Func& f);
     void apply(FuncC& f) const;
 
+    /**
+     * @brief Verify the table is set and set a database if one is not set. Recurses to all join refs.
+     *
+     * @throws If an empty string is passed for default then this will throw if the value is not set in the
+     *         instance.
+     *
+     * @param defaultDb the default database to assign, or an empty string for no default.
+     */
+    void verifyPopulated(std::string const& defaultDb="");
```

Comment:
Also, sometimes it's safer to give the following default to the parameter's value:
```
std::string const& defaultDb=std::string()
```
to avoid getting into this trap:
```
std::string const& defaultDb=" "
```